src/core/generator/from_openapi.rs

```diff
@@ -0,0 +1,84 @@
+use std::collections::BTreeMap;
+
+use oas3::Spec;
+
+use crate::core::config::{Config, Enum, RootSchema, Type, Union, Upstream};
+
+#[derive(Default)]
+pub struct OpenApiToConfigConverter {
+    pub query: String,
+    pub spec: Spec,
+    pub types: BTreeMap<String, Type>,
+    pub unions: BTreeMap<String, Union>,
+    pub enums: BTreeMap<String, Enum>,
+}
+
+impl OpenApiToConfigConverter {
+    pub fn new(query: impl AsRef<str>, spec_str: impl AsRef<str>) -> anyhow::Result<Self> {
+        let spec = oas3::from_reader(spec_str.as_ref().as_bytes())?;
+        Ok(Self {
+            query: query.as_ref().to_string(),
+            spec,
+            ..Default::default()
+        })
+    }
+
+    pub fn convert(self) -> Config {
+        let config = Config {
+            server: Default::default(),
+            upstream: Upstream {
+                base_url: self.spec.servers.first().cloned().map(|server| server.url),
+                ..Default::default()
+            },
```

Comment:
don't set upstream while generating configurations. Set `base_url` in individual `@http` directives.

Reply:
done